src/config.coffee

```diff
@@ -798,6 +798,13 @@ class Config
     @setDefaults(keyPath, @extractDefaultsFromSchema(schema))
     @setScopedDefaultsFromSchema(keyPath, schema)
 
+    if keyPath
+      value = @get(keyPath)
+      unless value is undefined
+        @unset(keyPath) unless @set(keyPath, value)
```

Comment:
@benogle this was the way I came up with to fix https://github.com/atom/atom/issues/5219, not sure if there is a better way. Basically when a schema is set, unset the value if it is no longer valid.